src/adhocracy_core/adhocracy_core/scripts/auto_transition_process_workflow.py

```diff
@@ -0,0 +1,109 @@
+"""Automatically transition workflow states.
+
+This is registered as console script.
+"""
+import transaction
+import argparse
+import inspect
+import logging
+
+from pyramid.paster import bootstrap
+from pyramid.registry import Registry
+from substanced.util import find_service
+
+from adhocracy_core.authorization import create_fake_god_request
+from adhocracy_core.interfaces import IResource
+from adhocracy_core.interfaces import search_query
+from adhocracy_core.resources.process import IProcess
+from adhocracy_core.sheets.workflow import IWorkflowAssignment
+from adhocracy_core.workflows import transition_to_states
+from adhocracy_core.utils import now
+
+
+logger = logging.getLogger(__name__)
+
+
+def auto_transition_process_workflow():  # pragma: no cover
+    """Automatically transition workflow states.
+
+    usage::
+
+        bin/auto_auto_transition_process_workflow etc/development.ini
+    """
+    docstring = inspect.getdoc(auto_transition_process_workflow)
+    parser = argparse.ArgumentParser(description=docstring)
+    parser.add_argument('ini_file',
+                        help='path to the adhocracy backend ini file')
+    args = parser.parse_args()
+    env = bootstrap(args.ini_file)
+    _auto_transition_process_workflow(env['root'], env['registry'])
+    transaction.commit()
+    env['closer']()
+
+
+def _auto_transition_process_workflow(context: IResource, registry: Registry):
+    catalogs = find_service(context, 'catalogs')
+    query = search_query._replace(interfaces=IProcess,
+                                  resolve=True
+                                  )
+    processes = catalogs.search(query).elements
+    for process in processes:
+        workflow_assignment = registry.content.get_sheet(process,
+                                                         IWorkflowAssignment)
+        if _workflow_auto_transition_enabled(registry, workflow_assignment) \
+            and _workflow_auto_transition_needed(process, registry,
+                                                 workflow_assignment):
+            _do_workflow_auto_transition(process, registry)
+
```

Comment:
Common styleguide hints::

For me the names are nicer to read/shorter without the "workflow" prefix, the context  is obivious. Also passing around the assingment sheet looks like needless performance optimization.

If statements should fit in one line and should check variables or very simple statements (easier to read and debug)

variable names  for booleans should be named postive: `has_xyz` or `is_xyz` (easier to read)

Functions should be short  and follow one abstraction level (at least for the first levels) (easier to read)

example refactoring first abstaction level (search/resources):
 
    ...
    has_auto_transition = auto_transition_enabled(process, registry)
    is_outdated = state_is_outdated(process, registry)
    if has_auto_transition and is_outdated:
    ...    do_auto_transition(process, registry)